mkt/websites/models.py

```diff
@@ -30,9 +33,70 @@ class Website(ModelBase):
     def get_fallback(cls):
         return cls._meta.get_field('default_locale')
 
+    @classmethod
+    def get_indexer(self):
+        return WebsiteIndexer
+
+    @property
+    def keywords(self):
+        return [t.tag_text for t in self._keywords.all()]
+
+    @property
+    def icon_url(self):
+        # XXX probably a different static location
+        return static_url('ADDON_ICON_URL') % (0, self.id, 128, self.icon_hash)
+
     def __unicode__(self):
         return unicode(self.url or '(no url set)')
 
 
+# Maintain ElasticSearch index.
+@receiver(models.signals.post_save, sender=Website,
+          dispatch_uid='website_index')
+def update_search_index(sender, instance, **kw):
+    instance.get_indexer().index_ids([instance.id])
+
+
+# Delete from ElasticSearch index on delete.
+@receiver(models.signals.post_delete, sender=Website,
+          dispatch_uid='website_unindex')
+def delete_search_index(sender, instance, **kw):
+    instance.get_indexer().unindex(instance.id)
+
+
+# Save translations before saving Website instance with translated fields.
 models.signals.pre_save.connect(save_signal, sender=Website,
                                 dispatch_uid='website_translations')
+
+
+class DBStore(object):
```

Comment:
I don't really see the point of this abstraction. If it's to be able to run tests with `MemoryStore`, it's dangerous because its behaviour won't necessarily match reality. We can already switch from ES to DB easily in a view by just switching the queryset / paginator (not mandatory, just for perf) / serializer (which can inherit from a base one).